osbs/build/build_request.py

```diff
@@ -502,6 +502,12 @@ def is_custom_base_image(self):
         return bool(re.match('^koji/image-build(:.*)?$',
                              self.base_image or ''))
 
+    def is_from_scratch_image(self):
+        """
+        Returns whether or not this is a build `FROM scratch`
+        """
+        return self.base_image == 'scratch'
```

Comment:
What happen if a developer will use `FROM scratch:latest` (because the one can)? May it cause weird errors like manifest for parent image not found, as I already experienced? Shouldn't we rather proceed as from scratch build and let it fail at the right step, or check it and fail early?

Reply:
```
Sending build context to Docker daemon 2.048 kB
Step 1/1 : FROM scratch:latest
'scratch' is a reserved name
```

Reply:
Yes, I tested that. But we have plugins that tries to download metadata(?) about parent before build.

Reply:
If it helps: this PR is about making specific 'FROM scratch' builds fail. @rcerven is working on making them work.